betty/user.py

```diff
@@ -2,8 +2,161 @@
 An API to interact with Betty's user.
 """
 
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import TYPE_CHECKING, Self, TypeVar, final, overload
+
+from betty.typing import Void
+
+if TYPE_CHECKING:
+    from types import TracebackType
+
+    from betty.assertion import Assertion
+    from betty.locale.localizable import Localizable
+    from betty.locale.localizer import Localizer
+
+
+_T = TypeVar("_T")
+
 
 class UserFacing:
     """
     A sentinel to mark something as being visible to users (e.g. not internal).
     """
+
+
+class Severity(Enum):
```

Comment:
Instead of severity we want to track verbosity so we're consistent with the console options